From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Revanth Kumar Uppala <ruppala@nvidia.com>
Date: Wed, 12 Oct 2022 12:04:24 +0530
Subject: [PATCH 036/240] NVIDIA: SAUCE: net: phy: aqr113c: Handling Duplicate
 WOL setting

Return Success on Duplicate WOL setting

Bug 3789538

Change-Id: Ie51ad5091a73dc16f22ffd99421767cdbd994a4b
Signed-off-by: Revanth Kumar Uppala <ruppala@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/linux-stable/+/2909753
Tested-by: Abhilash G <abhilashg@nvidia.com>
Reviewed-by: Abhilash G <abhilashg@nvidia.com>
Reviewed-by: Laxman Dewangan <ldewangan@nvidia.com>

diff --git a/drivers/net/phy/aquantia/aquantia_main.c b/drivers/net/phy/aquantia/aquantia_main.c
index f1f1a9d632465..1143efd50a91f 100644
--- a/drivers/net/phy/aquantia/aquantia_main.c
+++ b/drivers/net/phy/aquantia/aquantia_main.c
@@ -1433,7 +1433,7 @@ static int aqr113c_wol_settings(struct phy_device *phydev, bool enable)
 
 		/* restart auto-negotiation */
 		genphy_c45_restart_aneg(phydev);
-		priv->wol_status = 1;
+		priv->wol_status = WAKE_MAGIC;
 	} else {
 		/* Disable the WoL enable bit */
 		val = phy_read_mmd(phydev, MDIO_MMD_AN, MDIO_AN_RESVD_VEND_PROV);
@@ -1475,6 +1475,12 @@ static void aqr113c_get_wol(struct phy_device *phydev, struct ethtool_wolinfo *w
 
 static int aqr113c_set_wol(struct phy_device *phydev, struct ethtool_wolinfo *wol)
 {
+	struct aqr107_priv *priv = phydev->priv;
+
+	/* Return success if WOL is already set. Don't entertain duplicate setting of WOL */
+	if (!(priv->wol_status ^ wol->wolopts))
+		return 0;
+
 	if (wol->wolopts & WAKE_MAGIC)
 		return aqr113c_wol_settings(phydev, true);
 	else
-- 
2.43.0

